internal/services/mssqlmanagedinstance/mssql_managed_instance_start_stop_schedule_resource.go

```diff
@@ -0,0 +1,352 @@
+package mssqlmanagedinstance
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	schedule "github.com/hashicorp/go-azure-sdk/resource-manager/sql/2023-08-01-preview/startstopmanagedinstanceschedules"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type SqlManagedInstanceStartStopScheduleModel struct {
+	Name                 string              `tfschema:"name"`
+	SqlManagedInstanceId string              `tfschema:"managed_instance_id"`
+	Description          string              `tfschema:"description"`
+	ScheduleList         []ScheduleItemModel `tfschema:"schedule_list"`
+	TimeZoneId           string              `tfschema:"timezone_id"`
+	NextExecutionTime    string              `tfschema:"next_execution_time"`
+	NextRunAction        string              `tfschema:"next_run_action"`
+}
+
+type ScheduleItemModel struct {
+	StartDay  schedule.DayOfWeek `tfschema:"start_day"`
+	StartTime string             `tfschema:"start_time"`
+	StopDay   schedule.DayOfWeek `tfschema:"stop_day"`
+	StopTime  string             `tfschema:"stop_time"`
+}
+
+type MsSqlManagedInstanceStartStopScheduleResource struct{}
+
+var _ sdk.ResourceWithUpdate = MsSqlManagedInstanceStartStopScheduleResource{}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) ResourceType() string {
+	return "azurerm_mssql_managed_instance_start_stop_schedule"
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) ModelObject() interface{} {
+	return &SqlManagedInstanceStartStopScheduleModel{}
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.ManagedInstanceStartStopScheduleID
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
```

Comment:
Add a `name` argument with a validation func which only allows "default".

Reply:
I have confirmed with the service team that this resource will have 1to1 relationship with the `managedinstance` and hence the name will always be "default", I think leaving this out so that user doesn't need to declare it is cleaner.